### PR TITLE
Reghandler+armor base

### PR DESCRIPTION
**Pull request by m00nwtchr**
Done these 3 things:
https://trello.com/c/NWaq0SqI/8-base-armor-class
https://trello.com/c/sVCueR4X/2-add-a-registry-handler-for-items-and-blocks
https://trello.com/c/e2myWQT7/9-basic-armor
and also fixed build.gradle

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -23,26 +23,25 @@ compileJava {
...
     version = "${project.minecraft_version}-${project.forge_version}"
     runDir = "run"
-    
+
     // the mappings can be changed at any time, and must be in the following format.
     // snapshot_YYYYMMDD   snapshot are built nightly.
     // stable_#            stables are built at the discretion of the MCP team.
     // Use non-default mappings at your own risk. they may not always work.
     // simply re-run your setup task after changing the mappings to update your workspace.
-    mappings = project.forge_mappings
```
Comment (Edicatax) [inline build.gradle:32]: Why did you make this a hard mapping instead of simply changing the value in gradle.properties?

Comment (thebrightspark) [reply build.gradle:32]: I've already fixed this on master, so we will make sure that we use the changes on master for this file.

File: src/main/java/io/github/mdcteam/practice/items/ItemBasicArmor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        return createArmorSet(name, EnumHelper.addArmorMaterial(resLoc + "_material", resLoc.toString(), durability, reductions, enchantability, equipSound, toughness), equipmentSlots);
+    }
+
+    public static Set<ItemBasicArmor> createArmorSet(String name, ArmorMaterial material, EntityEquipmentSlot... equipmentSlots) {
+        Set<ItemBasicArmor> armorSet = new HashSet<>();
+        if(equipmentSlots.length == 0) equipmentSlots = new EntityEquipmentSlot[] {EntityEquipmentSlot.HEAD, EntityEquipmentSlot.CHEST, EntityEquipmentSlot.LEGS, EntityEquipmentSlot.FEET};
+        Set<EntityEquipmentSlot> slotSet = Sets.newHashSet(equipmentSlots);
+        slotSet.forEach(slot -> {
+            if (slot.getSlotType() != EntityEquipmentSlot.Type.ARMOR)
+                throw new IllegalArgumentException("Cannot create armour from the equipment slot type " + slot);
```
Comment (Edicatax) [inline src/main/java/io/github/mdcteam/practice/items/ItemBasicArmor.java:28]: Is this exception caught and handled anywhere or does minecraft/liblib/java do that implicitly and am I just super rusty?

Comment (thebrightspark) [reply src/main/java/io/github/mdcteam/practice/items/ItemBasicArmor.java:28]: I believe that MC/Forge will catch it and create an error report for errors on the main threads.
Tbh it should never be thrown - just using it to make sure people don't do silly things :)

**Pull request merged**